contents/python/introduction-to-python.md

```diff
@@ -4,25 +4,30 @@ Python is a simple yet powerful and versatile language. Conceived in the late 80
 
 ## Iterables
 
-[Iterables, Iterators and Generators](https://excess.org/article/2013/02/itergen1/)
+Strings, lists, and dictionaries belong to a type of class known as `Iterables`. If you're interested in delving nto the inner workings of `Iterable` classes, check out [Iterables, Iterators and Generators](https://excess.org/article/2013/02/itergen1/) by Ian Ward. He begins by explaining the `Iterable` class, then goes into the `Iterator` and `Generator` classes, both of which are powerful tools in Python.
 
-## Slicing
+## Indexing and Slicing
 
-[Introduction to indexing and slicing](https://samsontmr.github.io/Slicing-and-Dicing/)
+Accessing an element in a list using its index is known as indexing, e.g. `my_list[0]` returns the first element of my_list (Iterables are zero-indexed; this means that the first index is 0). Slicing, on the other hand, allows us to access a range of elements in a list. Extended slicing extends this functionality by introducing a step parameter, allowing us to do things like access every other element. Read more about this inn my blog post on [indexing and slicing](https://samsontmr.github.io/Slicing-and-Dicing/) in Python,
 
-## OOP in Python
+## Object Oriented Programming in Python
 
-[Python Class and OOP](https://jeffknupp.com/blog/2014/06/18/improve-your-python-python-classes-and-object-oriented-programming/)
+While Python is really useful for writing small scripts, often we want to use it to write whole applications. When this happens, having everything in one script (we call this procedural programming) will create huge problems readability-wise, especially when things start breaking and you're trying to find out why! This is where Object Oriented Programming comes in, offering a way to think about your program and organizing it into readable chunks. Jeff Knupp gives an introduction to [Python Classes and OOP](https://jeffknupp.com/blog/2014/06/18/improve-your-python-python-classes-and-object-oriented-programming/) here.
 
 
 ## Python for Data Science
 
-[Introduction to pandas](https://samsontmr.github.io/Sentimental-Pandas/)
-[Numpy Quickstart](https://docs.scipy.org/doc/numpy-dev/user/quickstart.html)
-[matplotlib Examples](http://matplotlib.org/examples/index.html)
-[Geetting Started with scikit-learn](http://scikit-learn.org/stable/tutorial/basic/tutorial.html)
-[scikit-learn Tutorials](http://scikit-learn.org/stable/tutorial/index.html)
\ No newline at end of file
+Data Science seems to be all the rage recently, so you'll be glad to know that Python has become a major player in the field of Data Science, rivalled only by R. Some commonly used Python libraries in Data Science include [pandas](http://pandas.pydata.org/) for data management and manipulation, [numpy](http://www.numpy.org/) and [scipy](http://www.scipy.org/) for scientific computing, [scikit-learn](http://scikit-learn.org/) for general machine learning and [matplotlib](http://matplotlib.org/) for data visualization.
```

Comment:
It is true that R and Python are very dominant in the scene, I just think that "only" is very strong. I might word it as "with its greatest rival being R". Using "only" sounds like there are no other players in the scene to me.